functional/sev/run.sh

```diff
@@ -0,0 +1,444 @@
+#!/bin/bash
+# Copyright 2022 Advanced Micro Devices, Inc.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+set -eE
+set -o nounset
+set -o pipefail
+
+trap cleanup EXIT
+
+GREEN='\033[0;32m'
+RED='\033[0;31m'
+NC='\033[0m'
+
+script_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+tests_repo_dir="$(realpath "$script_dir/../..")"
+
+export WORKING_DIR="${HOME}/work"
+
+export EDK2_REPO_DIR="${HOME}/edk2"
+export DOCKER_USER="rksavino"
+
+export GOROOT="/usr/local/go"
+export GOPATH="${WORKING_DIR}/go"
+export PATH="${PATH}:${GOROOT}/bin"
+
+export KBS_DB_HOST="127.0.0.1"
+export KBS_DB_USER="root"
+export KBS_DB_PW="root"
+export KBS_DB="simple_kbs"
+export KBS_DB_TYPE="mysql"
+
+esudo() {
+  sudo -E PATH=$PATH "$@"
+}
+
+build_and_install_kata_runtime_4270() {
+  cd ${WORKING_DIR}
+
+  # Clone kata-containers repo and checkout PR 4270
+  git clone https://github.com/kata-containers/kata-containers.git
+  cd kata-containers
+  gh pr checkout 4270
+
+  # Build and install the CC Kata Containers runtime
+  export GO111MODULE=on
+  cd ${WORKING_DIR}/kata-containers/src/runtime
+  make && esudo make install
+}
+
+build_rootfs() {
+  cd ${WORKING_DIR}/kata-containers/tools/osbuilder/rootfs-builder
+  esudo GOPATH=$GOPATH AGENT_INIT=yes USE_DOCKER=true DOCKER_RUNTIME=runc AA_KBC=offline_sev_kbc SKOPEO_UMOCI=yes ./rootfs.sh ubuntu
+
+esudo tee rootfs-ubuntu/etc/default/agent.cfg <<EOF
```

Comment:
I understand that this script is rebuilding the guest image just to have this `agent.cfg` file inside. We have a function (`cp_to_guest_img()`) which is used by the tests to copy files to the guest image so that this kind of rebuild is not needed.  This is the function https://github.com/kata-containers/tests/blob/CCv0/lib/common.bash#L255

So you can do the following:

1) place this file somewhere, for example, `functional/sev/fixtures/agent.cfg`
2) source the library file at the begin of this script (`${tests_repo_dir}/lib/common.bash`)
3) Before you execute the sev tests, use `cp_to_guest_img()` to copy the file to the guest image.

See how we do it in a test case for the agent: https://github.com/kata-containers/tests/blob/ab84d51927209ab9df8f144b63fbaaec8f8ef6c8/integration/containerd/confidential/agent_api.bats#L49

Reply:
Completed. I added a separate method in the `run.sh` to copy the agent.cfg. The one in common.bash is meant to work with dracut filesystem and the image isn't gunzipped. It uses `cat` during the export instead of `zcat` needed to export the gunzipped unbuntu initrd image.